Add programlisting nodes to content (#70)

vitaut opened the pull request:
The programlisting nodes and content nodes can interleave so we need to place them in the same array. Here's an example from the Doxygen docs:

https://www.stack.nl/~dimitri/doxygen/manual/examples/include/html/example.html

michaeljones commented:
Good catch! I had no idea that that was what was going on. Nice one for digging into it.

The `compound.py` is never going to be regenerated. They were automatically created but there have been so many manual tweaks since then either for minor improvements or just to cull out sections of unused code (it generates both reading & writing code) that we're never going to re-run the generation process. We could remove any comments that suggest otherwise.

vitaut commented:
Thanks for clarification. I've added a few comments and removed handling of `programlisting` since it is always empty now. If there are no objections, I'll merge this in.

michaeljones commented:
Looks great! Merge away.
